pkg/cloudevents/transport/x/doc.go

```diff
@@ -0,0 +1,49 @@
+/*
+
+Package transport/x shows proposed modifications to package transport.
+
+They are here for discussion purposes, anything that we can agree on
+should be merged into the transport package and may involve
+incompatible changes.
+
+Goals
+
+Support for intermediaries that receive and send events via different
+transports, for example adapters, importers, brokers or channels.
+
+Support for reliable messaging qualities of service between different
+transports
+
+Provide the option of blocking Receive vs. callback SetReceiver.
+There are many situations where the blocking style is easier to
+implement and work with.
+
+Overview of changes
+
+Sender/Receiver interfaces that can be implemented and used separately.
+Implementations can easily be combined into a single Transport that does both
+(e.g. for use by the client package)
+
+Message.Event() allows conversion from any message to the common
```

Comment:
I don't think you can do this without understanding both the transport and the codec.

Reply:
Correct - every Message implementation does understand the transport/codec it originated from - it is part of the transport implementation. Either Message.Event() would be implemented using  Codec.Decode() or vice-versa, either way the API is uniform.

Message.Event() == Codec.Decode(message)

The reason for introducing Message.Event() is to allow a Sender to decode  a message that originated from a different Receiver transport without needing to know how it was encoded. This also supports other Message implementations like EventMessage which is simply a Message interface wrapper for a locally-created Event object.